### ISSUE TITLE
Support for TCEFORM and TCAdefaults

### ISSUE THREAD

**Issue by PeaceSfiSh** [jweiland-net/maps2#228]
I'd like to restrict the `tx_maps2_domain_model_poicollection.collection_type` to **Point** only and added some TSconfig to a page:

```
TCAdefaults.tx_maps2_domain_model_poicollection.collection_type = Point
TCEFORM.tx_maps2_domain_model_poicollection.collection_type {
	keepItems = Point
}
# or
TCEFORM.tx_maps2_domain_model_poicollection.collection_type {
	disabled = 1
}
```

In this case the 'configuration_map' doesn't work, if creating a new record. No pointer is set and the use isn't able to set one.
This is because the setup of 'configuration_map' is based on an existing database record in 'tx_maps2_domain_model_poicollection'. Since no save and reload of the form is neccessary, with the default value already set, the record has the state 'NEW' without an uid set.

To solve this the 'configuration_map' always should be set up with a minimum of required parameters.
I'll gonna provide a pull request as a suggestion.

**Comment by froemken** [comment]
Good morning @PeaceSfiSh 

thank you for making maps2 a better extension. Tomorrow I will have a closer look into your code, but I'm unsure if it will made it into maps2. I'm sure it will work for you, but there are some internals which have to be respected, too.

In PoiCollection type "route" and "area" you can add further POIs with mouse click. This click creates an AJAX request and stores the POI directly to the PoiCollection. That's why I need an UID and can't work with NEW. Yes, default latitude/longitude will work for type "point" and "radius", but in case of "route" and "area" it represents the middle of the map and not a starting point of the polygon.

To get a fast solution for you, you should add PoiCollection type "Empty" to your keepItems list. Yes, I know, your editor needs an additional click on "Point" type.

Hopefully I can told you more, tomorrow.

Stefan

**Comment by PeaceSfiSh** [comment]
Thank you for your fast reply. It would be nice if you take a look in a timely manner. 

Keeping Empty as an option is possible too, because creating a maps isn't a daily task. But with your explanations I have two things in mind. First, inline records have to deal with the same problem, but I don't know the mechanisms exactly. The second idea is to store POIs in a hidden field in Json format and save them together with the record. This sounds like a lot of work just for using the settings up there. 😅  I would give this a low priority.

**Comment by froemken** [comment]
I will close this issue as it is solved with branch typo3_11_compatibility. If I'm done it will be merged in dev-master and version 10.0.0 will be released.
